packages/react-core/__tests__/operations/groupby.test.js

```diff
@@ -24,125 +25,97 @@ const COLUMN = 'test';
 
 describe('groupValuesByColumn', () => {
   test('should return null due to empty data array', () => {
-    expect(groupValuesByColumn([])).toEqual(null);
+    expect(groupValuesByColumn({ data: [] })).toEqual(null);
   });
 
   describe('valid features', () => {
     test('should return an empty array due to invalid operation', () => {
       expect(
-        groupValuesByColumn(
-          buildValidFeatures(COLUMN),
-          `${COLUMN}_quantitative`,
-          `${COLUMN}_qualitative`,
-          'pow'
-        )
+        groupValuesByColumn({
+          data: buildValidFeatures(COLUMN),
+          valuesColumns: [`${COLUMN}_quantitative`],
+          keysColumn: `${COLUMN}_qualitative`,
+          // @ts-ignore
+          operation: 'pow'
+        })
       ).toEqual([]);
     });
 
-    test(AggregationTypes.COUNT, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.COUNT
-      );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 3
-        },
-        {
-          name: 'Category 1',
-          value: 2
-        }
-      ]);
-    });
+    const RESULTS = {
+      [AggregationTypes.COUNT]: [
+        { name: 'Category 2', value: 3 },
+        { name: 'Category 1', value: 2 }
+      ],
+      [AggregationTypes.AVG]: [
+        { name: 'Category 2', value: 3 },
+        { name: 'Category 1', value: 3 }
+      ],
+      [AggregationTypes.MIN]: [
+        { name: 'Category 2', value: 1 },
+        { name: 'Category 1', value: 2 }
+      ],
+      [AggregationTypes.MAX]: [
+        { name: 'Category 2', value: 5 },
+        { name: 'Category 1', value: 4 }
+      ],
+      [AggregationTypes.SUM]: [
+        { name: 'Category 2', value: 9 },
+        { name: 'Category 1', value: 6 }
+      ]
+    };
 
-    test(AggregationTypes.AVG, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.AVG
-      );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 3
-        },
-        {
-          name: 'Category 1',
-          value: 3
-        }
-      ]);
+    describe('one valuesColumns', () => {
+      Object.entries(RESULTS).forEach(([operation, result]) => {
+        test(operation, () => {
+          const groups = groupValuesByColumn({
+            data: buildValidFeatures(COLUMN),
```

Comment:
If we always use same set of 'data' features for tests, using `buildValidFeatures(COLUMN)`... Doesn't it make sense to extract into a variable outside?

Reply:
Indeed, if we always use same 'COLUMN' (='test'), why don't we simplify and have a simple array of features, without any function?

Reply:
Improved.